dvc/fs/path.py

```diff
@@ -1,34 +1,44 @@
+import ntpath
 import posixpath
 
 
 class Path:
-    """
-    Class for operations on simple string paths.
+    def __init__(self, sep):
+        if sep == posixpath.sep:
+            self.flavour = posixpath
+        elif sep == ntpath.sep:
+            self.flavour = ntpath
+        else:
+            raise ValueError(f"unsupported separator '{sep}'")
```

Comment:
have to do it this way because dvcfs/repofs/gitfs are still using os.path as sep 🙁 They will be adjusted in the future and `fs.path` will be revisited as well.